src/main/java/com/amazon/dlic/auth/ldap/util/ConfigConstants.java

```diff
@@ -82,6 +82,9 @@ public final class ConfigConstants {
 
     public static final String LDAP_POOL_TYPE = "pool.type";
 
+    public static final String LDAP_POOL_PRUNING_PERIOD = "pool.pruning_period";
```

Comment:
Thank you for creating a ConfigConstant out of these values. It looks like there's a lack in documentation for these settings: https://github.com/opensearch-project/documentation-website/search?q=pruning.period and https://github.com/opensearch-project/documentation-website/search?q=pruning.idleTime.

I will create an issue on the documentation website to review the LDAP documentation and add or update where needed. 

While `lower_snake_case` appears to be the convention for settings keys, this may have the potential to break existing configs referencing these settings. 

What do you think about supporting the existing settings keys as well?

```
this.settings.getAsLong(ConfigConstants.LDAP_POOL_PRUNING_PERIOD, this.settings.getAsLong("pruning.period", 5l))
this.settings.getAsLong(ConfigConstants.LDAP_POOL_IDLE_TIME, this.settings.getAsLong("pruning.idleTime", 10l))
```

Reply:
@cwillum - Looks like the documentation site for LDAP Configuration needs a review. https://opensearch.org/docs/latest/security-plugin/configuration/ldap/

Reply:
> What do you think about supporting the existing settings keys as well?

Yes, good idea. I'll update it on Monday.

Reply:
> @cwillum - Looks like the documentation site for LDAP Configuration needs a review. https://opensearch.org/docs/latest/security-plugin/configuration/ldap/

The whole ldap2 backend type isn't documented, I'm not even sure if it's actually supposed to be used.

Reply:
I added an issue for review and update of LDAP documentation: [#1248](https://github.com/opensearch-project/documentation-website/issues/1248).